tests/thread_float/tests/01-run.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+
+#  Copyright (C) 2021 Otto-von-Guericke-Universität Magdeburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+# @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+
+import sys
+from testrunner import run
+
+
+def assertAlmostEqual(first, second, delta=0.05):
+    assert first + delta > second
+    assert first - delta < second
+
+
+def testfunc(child):
+    child.expect("THREADS CREATED")
+    child.expect("")
+    child.expect("THREAD t1 start")
+    child.expect("THREAD t2 start")
+    child.expect("THREAD t3 start")
+
+    child.expect(r"t(\d): (\d{3}\.\d+)")
+    first_thread = int(child.match.group(1))
+    # Note: intentionally keeping the float output as string to also test that printf("%f", ...)
+    # prints the exact same char sequence for the same float value each time
+    first_result = child.match.group(2)
+    child.expect(r"t(\d): (\d{3}\.\d+)")
+    second_thread = int(child.match.group(1))
+    second_result = child.match.group(2)
+
+    assert (first_thread == 1) or (first_thread == 3), "only thread t1 and t3 should print"
+    assert (second_thread == 1) or (second_thread == 3), "only thread t1 and t3 should print"
+    assert first_thread != second_thread, "t1 and t3 should print in turns"
+    if (first_thread == 1):
+        assertAlmostEqual(float(first_result), 141.44372894635987)
+        assertAlmostEqual(float(second_result), 141.44372894635987)
+    else:
+        assertAlmostEqual(float(second_result), 141.44372894635987)
+        assertAlmostEqual(float(first_result), 141.44372894635987)
```

Comment:
It seems the checks are the same in each block, just done in reversed order.

Reply:
indeed. Copy paste error, the second number should differ.

Reply:
Maybe use a `zip()` loop in that case. Prevents copy-pasta ;-)
```suggestion
    if (first_thread == 1):
        results = [float(first_result), float(second_result)]
    else:
        results = [float(second_result), float(first_result)]
    for result, expected in zip(results, [141.44372894635987, second_number]):
        assertAlmostEqual(result, expected)
```

Reply:
(of course you need to replace `second_number` with whatever is actually expected ;-))

Reply:
Thanks for pointing out the `zip()` loop. I decided to just recompute the expected result in python, since the duration of the test won't be affected by this (bound by the flops of the MCU or the speed of the UART / USB CDC ECM). This will also get rid of the magic numbers.